ts/components/wallet/WalletLayout.tsx

```diff
@@ -0,0 +1,150 @@
+import { Body, Container, Content, Text, View } from "native-base";
+import { Button } from "native-base";
+import { Left } from "native-base";
+import * as React from "react";
+import { ScrollView } from "react-native";
+import { Image, StyleSheet, TouchableHighlight } from "react-native";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+import I18n from "../../i18n";
+import ROUTES from "../../navigation/routes";
+import Icon from "../../theme/font-icons/io-icon-font";
+import variables from "../../theme/variables";
+import { WalletStyles } from "../styles/wallet";
+import AppHeader from "../ui/AppHeader";
+
+const styles = StyleSheet.create({
+  darkGrayBg: {
+    backgroundColor: variables.brandDarkGray
+  },
+  whiteBg: {
+    backgroundColor: variables.colorWhite
+  },
+  noBottomPadding: {
+    paddingBottom: 0
+  }
+});
+
+export enum CardType {
+  NONE,
+  FAN,
+  FULL,
+  HEADER
+}
+
+type Props = Readonly<{
+  title: string;
+  navigation: NavigationScreenProp<NavigationState>;
+  headerContents?: React.ReactNode;
+  cardType?: CardType;
+  showPayButton?: boolean;
+  allowGoBack?: boolean;
+}>;
+
+export class WalletLayout extends React.Component<Props> {
+  public static defaultProps = {
+    headerContents: null,
+    cardType: CardType.NONE,
+    showPayButton: true,
+    allowGoBack: true
+  };
+
+  private getLogo(): React.ReactNode {
+    switch (this.props.cardType) {
+      case CardType.FAN: {
+        return (
+          <View style={WalletStyles.container}>
+            <TouchableHighlight
+              onPress={(): boolean =>
+                this.props.navigation.navigate(ROUTES.WALLET_CREDITCARDS)
+              }
+            >
+              <Image
+                style={WalletStyles.pfCards}
+                source={require("../../../img/wallet/creditcards.jpg")}
+              />
+            </TouchableHighlight>
+          </View>
+        );
+      }
+      case CardType.FULL: {
+        return (
+          <View style={WalletStyles.container}>
+            <Image
+              style={{
+                height: 120,
+                width: "100%",
+                bottom: -1,
+                resizeMode: "contain"
+              }}
+              source={require("../../../img/wallet/card-tab.png")}
+            />
+          </View>
+        );
+      }
+      case CardType.HEADER: {
+        return (
+          <View style={WalletStyles.container}>
+            <Image
+              style={{
+                height: 55,
+                width: "100%",
+                bottom: -1,
+                resizeMode: "stretch"
+              }}
+              source={require("../../../img/wallet/single-tab.png")}
+            />
+          </View>
+        );
```

Comment:
for making sure to cover all cases add:

```ts
default:
  const neverCase: never = this.props.cardType;
  return null;
```

Reply:
I'm not sure I understand this adoption of the default case: it should cover the cases where cardType is `NONE`, but I don't get the `neverCase` part.

Other than that, am I correct in interpreting your comment as a different way of specifying the fallback behavior that is currently implemented as the `return null` after the switch? Is it because of readability?

Reply:
yes exactly, see an explanation [here](https://stackoverflow.com/questions/39419170/how-do-i-check-that-a-switch-block-is-exhaustive-in-typescript)